tests/cases/compiler/noUncheckedIndexAccess.ts

```diff
@@ -0,0 +1,21 @@
+//@noUncheckedIndexedAccess: true
+//@strictNullChecks: true
+
+enum Meat {
+    Sausage,
+    Bacon
+  }
+  const sausage = Meat.Sausage
+  const valueSausage = Meat[sausage]
+
+  const bacon = Meat.Bacon
+  const valueBacon = Meat[bacon]
+
+  const union: Meat.Bacon | Meat.Sausage = Meat.Bacon
+  const valueUnion = Meat[union]
+
+  //Avoiding a false positive
+  const value = Meat[0]
+
+  const t = "testing"
+  const value2 = Meat[t]

```

Comment:
`t` variable name should be self-explaining

Reply:
Made some changes, please have a look!